Firestore/core/test/unit/local/local_store_test.cc

```diff
@@ -1454,6 +1479,195 @@ TEST_P(LocalStoreTest, HandlesPatchMutationWithTransformThenRemoteEvent) {
+  AllocateQuery(query);
+
+  WriteMutation(testutil::PatchMutation(
+      "foo/bar", Map(), {testutil::Increment("sum", Value(1))}));
+
+  FSTAssertNotContains("foo/bar");
+  FSTAssertChanged(DeletedDoc("foo/bar"));
+
+  ApplyBundledDocuments({Doc("foo/bar", 1, Map("sum", 1337))});
+  FSTAssertChanged(
+      Doc("foo/bar", 1, Map("sum", 1), DocumentState::kLocalMutations));
+  FSTAssertContains(
+      Doc("foo/bar", 1, Map("sum", 1), DocumentState::kLocalMutations));
+
+  DocumentKeySet expected_keys({Key("foo/bar")});
+  FSTAssertQueryDocumentMapping(4, expected_keys);
+}
+
+TEST_P(LocalStoreTest, HandlesSavingAndCheckingBundleMetadata) {
+  BundleMetadata metadata("bundle", 1, SnapshotVersion(Timestamp(3, 0)));
+  EXPECT_FALSE(local_store_.HasNewerBundle(metadata));
+
+  local_store_.SaveBundle(metadata);
+
+  EXPECT_TRUE(local_store_.HasNewerBundle(metadata));
+}
+
+TEST_P(LocalStoreTest, HandlesSavingAndLoadingNamedQueries) {
+  core::Target target = Query("foo").ToTarget();
+
+  NamedQuery named_query(
+      "testQuery",
+      bundle::BundledQuery(std::move(target), core::LimitType::First),
+      SnapshotVersion(Timestamp::Now()));
+  local_store_.SaveNamedQuery(named_query, DocumentKeySet());
+
+  EXPECT_EQ(local_store_.GetNamedQuery("testQuery"), named_query);
+}
+
+TEST_P(LocalStoreTest,
+       SavingNamedQueriesAllocatesTargetsAndUpdatesTargetDocumentMapping) {
+  ApplyBundledDocuments({Doc("foo1/bar", 1, Map("sum", 1337)),
+                         Doc("foo2/bar", 1, Map("sum", 42))});
+  FSTAssertChanged(Doc("foo1/bar", 1, Map("sum", 1337)),
+                   Doc("foo2/bar", 1, Map("sum", 42)));
+  FSTAssertContains(Doc("foo1/bar", 1, Map("sum", 1337)));
+  FSTAssertContains(Doc("foo2/bar", 1, Map("sum", 42)));
+
+  core::Target target_1 = Query("foo1").ToTarget();
+
+  NamedQuery named_query_1(
+      "query-1",
+      bundle::BundledQuery(std::move(target_1), core::LimitType::First),
+      SnapshotVersion(Timestamp::Now()));
+  DocumentKeySet mapped_keys_1({Key("foo1/bar")});
+  local_store_.SaveNamedQuery(named_query_1, mapped_keys_1);
+
+  EXPECT_EQ(local_store_.GetNamedQuery("query-1"), named_query_1);
+  FSTAssertQueryDocumentMapping(4, mapped_keys_1);
+
+  core::Target target_2 = Query("foo2").ToTarget();
+
+  NamedQuery named_query_2(
+      "query-2",
+      bundle::BundledQuery(std::move(target_2), core::LimitType::First),
+      SnapshotVersion(Timestamp::Now()));
+  DocumentKeySet mapped_keys_2({Key("foo2/bar")});
+  local_store_.SaveNamedQuery(named_query_2, mapped_keys_2);
+
+  EXPECT_EQ(local_store_.GetNamedQuery("query-2"), named_query_2);
+  FSTAssertQueryDocumentMapping(6, mapped_keys_2);
+}
+
+TEST_P(LocalStoreTest, HandlesSavingAndLoadingLimitToLastQueries) {
+  core::Target target =
+      Query("foo")
+          .AddingOrderBy(testutil::OrderBy(testutil::Field("length"),
+                                           core::Direction::Descending))
+          // Use LimitToFirst here so `ToTarget()` does
```

Comment:
Nit: the comment seems to need reformatting, line width is too narrow.

Reply:
Fixed.